Menu/PageTreeMenuBuilder.php

```diff
@@ -72,6 +72,12 @@ private function appendNodes (ItemInterface $parent, array $nodes, array $option
     {
         foreach ($nodes as $node)
         {
+            // Don't render nodes that the User is not supposed/eligible to see
```

Comment:
This will produce the same issue with the menu not being able to find the current route (if it is hidden). And therefore not being able to highlight the other nodes as "current_ancestor"?

Your comment ist technically wrong, as we are building the data structure here, not the rendering.
So we probably should leave this code here as is and just skipping the nodes when rendering?